schema-engine/connectors/sql-schema-connector/src/flavour/postgres/renderer.rs

```diff
@@ -483,7 +484,11 @@ impl SqlRenderer for PostgresRenderer {
             let table = tables.previous.name();
 
             for index in tables.previous.indexes().filter(|idx| !idx.is_primary_key()) {
-                result.push(self.render_drop_index(index));
+                let rendered_ns = index
+                    .table()
+                    .namespace()
+                    .filter(|&ns| !is_default_namespace_in_schema(ns, schemas.next));
+                result.push(self.render_drop_index(rendered_ns, index));
```

Comment:
It's not ideal that it's not consistent with the other renderer methods, but the only way to make it consistent would be to pass the `SqlSchema` or `MigrationPair` everywhere (and maybe introduce extension methods like `namespace_if_not_default(schema)` to walkers to reduce boilerplate), which doesn't sound very good either.

I don't have any other ideas here, and it's not a blocking comment.